### PR TITLE
Remove uri slot from Entity REF

### LINKED ISSUE #241: Conversion to JSON-LD is not working as expected
When converting the SSSOM table to JSON-LD, the slots defined as `EntityReference` have the range `rdfs:Resource` as the datatype. So for JSON-LD, it's just another string that doesn't recognise the prefixes.
In these cases, it should have the `@type: @id`. For this, the range in the schema should be `uri`.

Here is a piece of the JSON-LD `@context`

```json
"@context": {
    ...
    "subject_id": {
        "@type": "rdfs:Resource",
        "@id": "owl:annotatedSource"
      },
    "predicate_id": {
        "@type": "rdfs:Resource",
        "@id": "owl:annotatedProperty"
      },
    "object_id": {
        "@type": "rdfs:Resource",
        "@id": "owl:annotatedTarget"
      },
    "mapping_justification": {
        "@type": "rdfs:Resource"
      },
    ...
}
    
```

Here is how it's loaded in RDF4J

Screenshot 2022-11-25 at 12 00 19

After changing the type to `@id` (except for subject_id), this is how it looks in RDF4J

Screenshot 2022-11-22 at 16 17 34

There's also another issue with `Mapping`. It's not adding the `@type: Mapping` for each mapping. In the `@context`, there's the definition of Mapping.

```json
"@context": {
    ...
    "Mapping": {
       "@id": "owl:Axiom"
    }
    ...
}
```

```json
"mappings": [
      {
        "subject_id": "MP:0001289",
        "predicate_id": "skos:closeMatch",
        "object_id": "HP:0007968",
        "mapping_justification": "semapv:LexicalMatching",
        "subject_label": "persistence of hyaloid vascular system",
        "object_label": "Remnants of the hyaloid vascular system",
      },
      {
        "subject_id": "MP:0001293",
        "predicate_id": "skos:exactMatch",
        "object_id": "HP:0000528",
        "mapping_justification": "semapv:LexicalMatching",
        "subject_label": "anophthalmia",
        "object_label": "Anophthalmia",
      },
      {
        "subject_id": "MP:0001303",
        "predicate_id": "skos:closeMatch",
        "object_id": "HP:0000517",
        "mapping_justification": "semapv:LexicalMatching",
        "subject_label": "abnormal lens morphology",
        "object_label": "Abnormality of the lens",
      }
]
```

### PR DESCRIPTION
**Pull request by matentzn**
Fixes #241 

trying to model the Entity reference correctly

Fails:

https://github.com/mapping-commons/sssom/pull/244/files#diff-
5607ecbac7dd39511f45b6ae9cd7bf6140c32d1c61674e594c09468256c7e17fR20

### CODE REVIEW

**Comment by matentzn** [comment]
Think about adding a test here.

**Comment by matentzn** [comment]
See also here: https://github.com/mapping-commons/sssom/issues/241#issuecomment-1332728870

File: project.Makefile [hunk 1]
```diff
@@ -10,4 +10,20 @@ build:
...
-	poetry publish
\ No newline at end of file
+	poetry publish
+
+tmp/sssom.py:
+	$(MAKE) src/sssom_schema/datamodel/sssom_schema.py
+	mv 
+
+tests/output/out.json: $(SOURCE_SCHEMA_PATH) tests/input/mapping_set.ttl
+	$(RUN) linkml-convert -f ttl -C MappingSet -t json -s $^ --output $@
```
Comment (matentzn) [inline project.Makefile:20]: This is the line that currently fails.

**Comment by matentzn** [comment]
@hrshdhgd can you help me with the PR again when you get a minute?

I cant get https://github.com/mapping-commons/sssom/pull/244/files#diff-5607ecbac7dd39511f45b6ae9cd7bf6140c32d1c61674e594c09468256c7e17fR20 to work..

File: project.Makefile [hunk 2]
```diff
@@ -10,4 +10,20 @@ build:
...
+
+tests/output/out.ttl: $(SOURCE_SCHEMA_PATH) tests/output/out.json
+	$(RUN) linkml-convert -f json -C MappingSet -t ttl -s $^ --output $@
+
+tests/output/out.jsonld: $(SOURCE_SCHEMA_PATH) tests/output/out.ttl tests/input/context.jsonld
+	$(RUN) linkml-convert -f ttl -C MappingSet -t json-ld -c tests/input/context.jsonld -s $(SOURCE_SCHEMA_PATH) tests/output/out.ttl --output $@
+
+test-convert: tests/output/out.jsonld
```
Comment (matentzn) [inline project.Makefile:28]: @hrshdhgd does this here work?

Comment (hrshdhgd) [reply project.Makefile:28]: Yes

**Comment by hrshdhgd** [comment]
Found the solution. `linkml` had a line missing. I have a PR for this: https://github.com/linkml/linkml/pull/1153

**Comment by matentzn** [comment]
THANK YOU @hrshdhgd, much appreciated. Once this is merged, is it your understanding the the jsonld export will work in general?

**Comment by matentzn** [comment]
No longer relevant or needs to be totally rethought.

**Pull request closed without merging**